02-Arrays/02-Substitution/substitution.c

```diff
@@ -0,0 +1,145 @@
+#include <cs50.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+
+bool validate_arguments(int arguments);
+bool validate_key(string key);
+bool validate_key_length(string key);
+bool validate_key_alphabetic(string key);
+bool validate_key_unique(string key);
+char encrypt_character(char character, string key);
+
+
+int main(int argc, string argv[])
+{
+    bool arguments_valid = validate_arguments(argc);
+    if (arguments_valid != true)
+    {
+        return 1;
+    }
+
+    bool key_valid = validate_key(argv[1]);
+    if (key_valid != true)
+    {
+        return 1;
+    }
+
+    string plaintext = get_string("plaintext: ");
+    string key = argv[1];
+
+    printf("ciphertext: ");
+
+    for (int i = 0; i < strlen(plaintext); i++)
+    {
+        char encrypted_character = encrypt_character(plaintext[i], key);
+        printf("%c", encrypted_character);
+    }
+    printf("\n");
+
+    return 0;
+
+}
+
+char encrypt_character(char character, string key)
+{
+    if (isalpha(character))
+    {
+        bool is_upper = isupper(character);
+
+        character = toupper(character);
+        char first_alphabet_letter = 'A';
+        int index = character - first_alphabet_letter;
+        char encrypted_character = key[index];
+
+        if (!is_upper)
+        {
+            encrypted_character = tolower(encrypted_character);
+        }
+
+        return encrypted_character;
+    }
+    else
+    {
+        return character;
+    }
+}
+
+bool validate_arguments(int arguments)
+{
+    bool valid = true;
+
+    if (arguments != 2)
+    {
+        printf("Usage: ./substitution KEY\n");
+        valid = false;
+    }
+
+    return valid;
```

Comment:
For this specific code not so relevant, but in general:

Be careful of defaulting to valid.
If you have a bug anywhere (for example in the if condition), then the arguments will be marked as valid.
By setting the default to **not** valid, you say: it's only valid when xyz

Imagine a bug in the password validation and defaulting to the password being valid 😉